packages/main/src/plugin/extension-loader.ts

```diff
@@ -312,27 +314,47 @@ export class ExtensionLoader {
       // collect all extensions from the pluginDirectory folders
       const analyzedPluginsDirectoryExtensions: AnalyzedExtension[] = (
         await Promise.all(pluginDirectories.map(folder => this.analyzeExtension(folder, true)))
-      ).filter(extension => extension !== undefined) as AnalyzedExtension[];
+      ).filter(extension => !extension.error) as AnalyzedExtension[];
       analyzedExtensions.push(...analyzedPluginsDirectoryExtensions);
     }
 
     // now we have all extensions, we can load them
     await this.loadExtensions(analyzedExtensions);
   }
 
-  async analyzeExtension(extensionPath: string, removable: boolean): Promise<AnalyzedExtension | undefined> {
+  async analyzeExtension(extensionPath: string, removable: boolean): Promise<AnalyzedExtension> {
     // do nothing if there is no package.json file
+    let error = undefined;
+    const disposables: Disposable[] = [];
     if (!fs.existsSync(path.resolve(extensionPath, 'package.json'))) {
-      console.warn(`Ignoring extension ${extensionPath} without package.json file`);
-      return undefined;
+      error = `Ignoring extension ${extensionPath} without package.json file`;
+      console.warn(error);
+      const analyzedExtension: AnalyzedExtension = {
+        id: '<unknown>',
+        name: '<unknown>',
+        path: extensionPath,
+        manifest: undefined,
+        api: <typeof containerDesktopAPI>{},
+        removable,
+        subscriptions: disposables,
+        dispose(): void {
+          disposables.forEach(disposable => disposable.dispose());
+        },
```

Comment:
I'm not sure why we dispose the disposables. Do we register subscriptions in case of errors ?

Reply:
Because we still have to return the subscriptions array from here I felt like we should correctly dispose it... but had forgot it was readonly anyway. Simplified this in new commit.